src/lib.rs

```diff
@@ -738,7 +738,7 @@ fn set(
 ) -> Result<(), String> {
     let actions_str = actions;
     // `actions` are in the format of `failpoint[->failpoint...]`.
-    let actions = try!(actions.split("->").map(Action::from_str).collect());
+    let actions = r#try!(actions.split("->").map(Action::from_str).collect());
```

Comment:
Why this one doesn't become `action.split("->").map(Action::from_str).collect()?` like lines 502 and 507?

Reply:
Type inference failure:

```
error[E0284]: type annotations required: cannot resolve `<_ as std::ops::Try>::Ok == _`
   --> src/lib.rs:741:19
    |
741 |     let actions = actions.split("->").map(Action::from_str).collect()?;
    |                   ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
```

`try!` and `?` work differently, where `?` requires more type inference, and the compiler can't figure out that we want `collect` to go through the [FromIterator impl that converts from an iterator over `Result` to a `Result` of a collection](https://doc.rust-lang.org/stable/std/iter/trait.FromIterator.html#impl-FromIterator%3CResult%3CA%2C%20E%3E%3E). This is probably because `try!` relies on concrete `Result` types, where `?` only requires impls of `Try`.

I've pushed a followup commit to remove this instance of the macro by adding a type hint.